File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/solution.cc
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        auto req = 0;
        const int N = blocks.size();
        for (int i = 0; i < k; ++i) {
            if (blocks[i] == 'W') {
                ++req;
            }
        }
        int min_req = req;
        for (int i = k; i < N; ++i) {
            if (blocks[i - k] == 'B') ++req;
            if (blocks[i] == 'B') --req;
            min_req = min(min_req, req);
        }

        return min_req;
    }
};

File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/solution.rs
impl Solution {
    pub fn minimum_recolors(blocks: String, k: i32) -> i32 {
        let mut w_cnt = 0;
        let k = k as usize;
        let blocks: Vec<char> = blocks.chars().collect();
        for i in 0..k {
            if blocks[i] == 'W' {
                w_cnt += 1
            }
        }
        let mut min = w_cnt;
        if min == 0 {
            return 0;
        }
        let n = blocks.len();
        for i in k..n {
            if blocks[i - k] == 'W' {
                w_cnt -= 1;
            }
            if blocks[i] == 'W' {
                w_cnt += 1;
            }
            min = std::cmp::min(min, w_cnt);
            if min == 0 {
                return 0;
            }
        }
        return min;
    }
}

